vsl-sdk/src/lib.rs

```diff
@@ -55,14 +86,66 @@ impl Amount {
             None
         }
     }
+
     pub const fn checked_add(self, rhs: Amount) -> Option<Amount> {
         if let Some(v) = self.0.checked_add(rhs.0) {
             Some(Amount(v))
         } else {
             None
         }
     }
+
+    pub fn to_hex_str(&self) -> String {
+        format!("{:#x}", self)
+    }
+
+    pub fn from_hex_str(s: &str) -> Result<Self, ParseAmountError> {
+        if !s.starts_with("0x") {
+            return Err(ParseAmountError::NotHex);
+        }
+        let s = &s[2..];
+        if s == "0" {
+            return Ok(Self::ZERO);
+        }
+        if s.starts_with('0') {
+            return Err(ParseAmountError::LeadingZeros);
+        } else {
+            let subunits =
+                u128::from_str_radix(s, 16).map_err(ParseAmountError::ParseInt)?;
+            return Ok(Self::from_subunits(subunits));
+        }
+    }
+
+    pub fn to_str_with_decimals(&self, decimals: u8) -> String {
+        let one_token = 10u128.checked_pow(decimals as u32).expect("overflow");
+        let units = self.0 / one_token;
+        let subunits = self.0 % one_token;
+        format!("{}.{:0>width$}", units, subunits, width = decimals as usize)
```

Comment:
This will print with trailing zeros out to the full precision. Is that actually standard anywhere or what we want?
Just a single source, etherscan prints in decimals in at least two units (ETH and Gwei) with no trailing zeros (random recent transaction:) https://etherscan.io/tx/0xe7bc332a582cf1aa80a8d87645d68d0c5e4f3c22c5a022c9950910fef2cd2bff